game/cards/dm04/guardian.go

```diff
@@ -38,5 +38,24 @@ func MistRiasSonicGuardian(c *match.Card) {
 	c.ManaCost = 5
 	c.ManaRequirement = []string{civ.Light}
 
-	c.Use(fx.Creature, fx.When(fx.AnotherCreatureSummoned, fx.MayDraw1))
+	c.Use(fx.Creature, fx.When(fx.Summoned, func(card *match.Card, ctx *match.Context) {
```

Comment:
Use `fx.InTheBattlezone` instead of `fx.Summoned` for both.

Reply:
That would be a bug. I presume it would create multiple persistent effects.

Reply:
The doc of the func: 
> // InTheBattlezone returns true if the card arrives in the battlezone.
> //
> // Similar to summon but activates also if the card was under an Evolution card and becomes visible again.
> // Used for cards that have continuous effects while in the battlezone.

Reply:
Hmm, I see.